### PR TITLE
music: add support for multiple file formats.

### PR DESCRIPTION
**Pull request by carlo-bramini**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This patch adds support for more audio formats when playing audio tracks.

### CODE REVIEW

File: src/game/music.c [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 #include "game/gameflow.h"
 #include "game/sound.h"
 #include "global/vars.h"
 #include "log.h"
+#include "memory.h"
 #include "specific/s_audio.h"
 
 #include <stdio.h>
 
+static const char *m_Extensions[] = { ".flac", ".ogg", "mp3", ".wav", NULL };
```
Comment (rr-) [inline src/game/music.c]: mp3 has no `.`.

File: src/game/music.c [hunk 2]
```diff
@@ -31,6 +35,15 @@ static void Music_StopActiveStream(void)
     S_Audio_StreamSoundClose(m_AudioStreamID);
 }
 
+static char *Music_GetTrackFileName(int track)
+{
+    char file_path[64];
+
+    sprintf(file_path, "music/track%02d.flac", track);
+
```
Comment (rr-) [inline src/game/music.c]: Unnecessary blank lines

File: src/game/music.c [hunk 3]
```diff
@@ -31,6 +35,15 @@ static void Music_StopActiveStream(void)
     S_Audio_StreamSoundClose(m_AudioStreamID);
 }
 
+static char *Music_GetTrackFileName(int track)
```
Comment (rr-) [inline src/game/music.c:39]: No declaration of this function at the start of the file

**Review by rr- — COMMENTED** [review]
Cosmetic remarks, other than that LGTM.

**Comment by rr-** [comment]
The commit message should spell `music` though without the caps.

**Review by rr- — APPROVED** [review]
LGTM

**Review by walkawayy — APPROVED** [review]

**Comment by rr-** [comment]
Actually this is missing a changelog entry.

**Review by rr- — APPROVED** [review]

**Pull request merged**